sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/basicLogicalOperators.scala

```diff
@@ -498,12 +498,16 @@ case class Sort(
 
 /** Factory for constructing new `Range` nodes. */
 object Range {
-  def apply(start: Long, end: Long, step: Long, numSlices: Option[Int]): Range = {
-    val output = StructType(StructField("id", LongType, nullable = false) :: Nil).toAttributes
+
+  def apply(start: Long, end: Long, step: Long, numSlices: Option[Int], outputName: Option[String])
```

Comment:
Why make this an option? It is not optional right? The next function should either call Range(start, end, step, Some(numSlices), "id") or this function should have a default parameter.